packages/app/src/systems/Error/guards/ReportErrorGuard.tsx

```diff
@@ -0,0 +1,18 @@
+import { useEffect } from 'react';
+import { Outlet } from 'react-router-dom';
+
+import { useReportError } from '../hooks';
+
+import { store } from '~/store';
+
+export function ReportErrorGuard() {
+  const { hasErrorsToReport } = useReportError();
+
+  useEffect(() => {
+    if (hasErrorsToReport) {
+      store.openViewReportErrors();
+    }
+  }, [hasErrorsToReport]);
+
+  return <Outlet />;
+}
```

Comment:
we can remove this guard and use the fetchMachine to open the reportErrors dialog?